src/System/FSNotify/Debouncer/Naive.hs

```diff
@@ -0,0 +1,111 @@
+{- |
+The Naive Debouncer aims to provide a quick and dirty debouncer logic. It does
+not try to do any kind of I/O. It also assumes that the events provided by the
+underlying library are in a coherent order.
```

Comment:
What does coherent mean? IIRC you can get some pretty strange event sequences on some platforms. I feel like we can remove this sentence, it should go without saying that a state machine like this should do its best to be sensible with the events it receives.

Reply:
What I was trying to convey is that, in contrast to the original proposal, the `NaiveDebouncer` will not try to detect any incoherent sequence of events, let's say `Added` followed by `Added`, and will not throw, ever. I will make a second pass on the comments to clarify this.